### PR TITLE
Clean up various algorithm-related tag things

### PR DESCRIPTION
**Pull request by wthrowe**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- Follows [code review guidelines](https://sxs-collaboration.github.io/spectre/code_review_guide.html)
- Code has documentation and unit tests
- Private member variables have a trailing underscore
- Do not use [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), e.g. `double* pd_blah` is bad
- Header order:
  1. hpp corresponding to cpp (only in cpp files) or `tests/Unit/TestingFramework.hpp` (only in tests)
  2. Blank line (only in cpp files and tests)
  3. STL and externals (in alphabetical order)
  4. Blank line
  5. SpECTRE includes (in alphabetical order)
- File lists in CMake are alphabetical
- Correct `noexcept` specification for functions (if unsure, mark `noexcept`)
- Mark objects `const` whenever possible
- Almost always `auto`, except with expression templates, i.e. `DataVector`
- All commits for performance changes provide quantitative evidence and the tests used to obtain said evidence.
- Make sure error messages are helpful, e.g. "The number of grid points in the matrix 'F' is not the same as the number of grid points in the determinant."
- Prefix commits addressing PR requests with `fixup`. These commits are flagged
  by Travis so we do not accidentally merge them.
- Include what you use, prefer forward declarations in hpp files.
- Explicitly make numbers floating point, e.g. `2.` or `2.0` over `2`
- Use `Tensor`'s non-member get if the indices are constant expressions, e.g.
  `get<1>(tensor)`

### CODE REVIEW

File: src/Evolution/DiscontinuousGalerkin/DgElementArray.hpp [hunk 1]
```diff
@@ -12,18 +14,33 @@
...
   using array_index = ElementIndex<volume_dim>;
 
+ private:
+  template <typename Action, typename = cpp17::void_t<>>
+  struct const_global_cache_tags {
+    using type = tmpl::list<>;
+  };
+
+  template <typename Action>
+  struct const_global_cache_tags<
```
Comment (kidder) [inline src/Evolution/DiscontinuousGalerkin/DgElementArray.hpp]: elevate this to somewhere that any component can use it

Comment (kidder) [reply src/Evolution/DiscontinuousGalerkin/DgElementArray.hpp]: add some documentation to the ParallelizationGroup

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by nilsvu** [comment]
Will this break code that uses the const global cache (or anything else)? If so, could you briefly describe what people that rebase on this need to pay attention to?

**Comment by wthrowe** [comment]
I believe the main breakage here will be if someone wrote a new executable using DgElementArray, where they will now have to change `dg_element_array_add_to_cache` to `const_global_cache_tag_list` in their metavariables, and preferably check whether some of the things in it should be removed.

I believe everything else is straightforward: SendDataForFluxes gained a template parameter, a few things I think are unlikely to have been used were removed as indicated in the commit messages (replacements are described there in prose, existing use cases can be checked for code versions).

**Comment by kidder** [comment]
@nilsdeppe this is ready for you to look at; if you are happy then Will can squash

**Review by nilsdeppe — DISMISSED** [review]
I'm assuming you guys verified the dox renders as expected :)

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@nilsdeppe this just needs your approval again after Will squashed

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**